packages/client/test/rpc/debug/setHead.spec.ts

```diff
@@ -18,31 +18,74 @@ describe(method, async () => {
     const exec = await testSetup(blockchain)
     await exec.run()
     const newHead = await (exec.vm.blockchain as Blockchain).getIteratorHead!()
-    assert.equal(newHead.header.number, BigInt(5), 'should run all blocks')
+    assert.equal(newHead.header.number, BigInt(mainnetData.length), 'should run all blocks')
 
-    const a = await createClient({ blockchain })
-    await a.service.skeleton?.open()
-    ;(a.service.execution as any) = exec
+    const client = await createClient({ blockchain })
+    await client.service.skeleton?.open()
+    ;(client.service.execution as any) = exec
 
-    const manager = createManager(a)
+    const manager = createManager(client)
     const rpc = getRPCClient(startRPC(manager.getMethods()))
     assert.equal(
-      await a.service.skeleton?.headHash(),
+      await client.service.skeleton?.headHash(),
       undefined,
       'should return undefined when head is not set',
     )
     for (let i = 0; i < blocks.length; i++) {
       await rpc.request(method, [`0x${i}`])
       assert.deepEqual(
-        await a.service.skeleton?.headHash()!,
+        await client.service.skeleton?.headHash()!,
         blocks[i].header.hash(),
         `skeleton chain should return hash of block number ${i} set as head`,
       )
       assert.deepEqual(
-        a.service.execution.chainStatus?.hash!,
+        client.service.execution.chainStatus?.hash!,
         blocks[i].header.hash(),
         `vm execution should set hash to new head`,
       )
     }
   }, 30000)
+
+  it('should return error for pending block', async () => {
+    const blockchain = await createBlockchainFromBlocksData(mainnetData, {
+      validateBlocks: true,
+      validateConsensus: false,
+    })
+
+    const client = await createClient({ blockchain })
+
+    const manager = createManager(client)
+    const rpc = getRPCClient(startRPC(manager.getMethods()))
+    const result = await rpc.request(method, ['pending'])
+    assert.equal(result.error.code, -32602)
+    assert.equal(result.error.message, '"pending" is not supported')
+  })
+
+  it('should handle internal errors', async () => {
+    const blockchain = await createBlockchainFromBlocksData(mainnetData, {
+      validateBlocks: true,
+      validateConsensus: false,
+    })
+    const blocks = await blockchain.getBlocks(0, 6, 0, false)
+    const exec = await testSetup(blockchain)
+    await exec.run()
+    const newHead = await (exec.vm.blockchain as Blockchain).getIteratorHead!()
+    assert.equal(newHead.header.number, BigInt(mainnetData.length), 'should run all blocks')
+
+    const client = await createClient({ blockchain })
+    ;(client.service.skeleton as any) = {
+      open: async () => {
+        throw new Error('open failed')
+      },
+    }
+
+    const manager = createManager(client)
+    const rpc = getRPCClient(startRPC(manager.getMethods()))
+
+    for (let i = 0; i < blocks.length; i++) {
```

Comment:
Can we just check 1 block here and not 6 and also only run one block above (just to make this particular test more efficient since the RPC tests in general are kind of slow and this will add more overall CI time)?

Reply:
addressed